### ISSUE TITLE
Misclassification of Cloud Shadows as Water Pixels in Sentinel-2 Scene Inference

### ISSUE THREAD

**Issue by flamboyant-vinci** [spaceml-org/ml4floods#113]
Description:
Issue Summary:
While running inference on a Sentinel-2 scene depicting flash flooding in Dubai using the ml4floods-1.0.1 library and pre-trained ml4floods model, cloud shadow pixels are misclassified as water. This misclassification is not observed in other scenes from the test images, such as the MSR264_18MIANDRIVAZODETAIL_DEL_v2 flood map from the test set of the WorldFloodsv2 dataset.

Steps to Reproduce:

1. Load the pre-trained Unet multioutput S2-to-L8 model from ml4floods huggingface repository
2. Run inference on a Sentinel-2 scene of flash flooding in Dubai. (S2A_MSIL1C_20240417T064631_N0510_R020_T40RCN_20240417T091941)
3. Observe that cloud shadow pixels are classified as water.

Expected Behavior:
The model should accurately distinguish between water pixels and cloud shadows, ensuring that only actual water bodies/inundated regions  are classified as such.

Actual Behavior:
Cloud shadow pixels are misclassified as water, leading to inaccurate flood maps.

Additional Information:
Trained model used: Unet multioutput S2-to-L8 in folder models/WF2_unetv2_bgriswirs.
Example image: [Sen2_dubai_flood_annotated](https://github.com/spaceml-org/ml4floods/assets/78052897/3dbda653-bd1b-4861-a869-193481e08115)

**Comment by gonzmg88** [comment]
Hi

Thanks for sharing, I'm sorry that the model doesn't work as expected in this scene. Urban flood segmentation is quite challenging. In general we observed that the model produce less false positives in cloud shadows in the data that we used for testing but this scene probes that there's still room for improvement. 

We're still working on flood segmentation so if we update the model we'll let you know.
